extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/dataportabilityproject/transfer/microsoft/calendar/MicrosoftCalendarExporter.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.transfer.microsoft.calendar;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.dataportabilityproject.spi.transfer.provider.ExportResult;
+import org.dataportabilityproject.spi.transfer.provider.Exporter;
+import org.dataportabilityproject.spi.transfer.types.ExportInformation;
+import org.dataportabilityproject.transfer.microsoft.transformer.TransformResult;
+import org.dataportabilityproject.transfer.microsoft.transformer.TransformerService;
+import org.dataportabilityproject.types.transfer.auth.TokenAuthData;
+import org.dataportabilityproject.types.transfer.models.calendar.CalendarContainerResource;
+import org.dataportabilityproject.types.transfer.models.calendar.CalendarEventModel;
+import org.dataportabilityproject.types.transfer.models.calendar.CalendarModel;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.dataportabilityproject.transfer.microsoft.transformer.TransformConstants.CALENDAR_ID;
+
+/**
+ * Exports Outlook calendar information using the Microsoft Graph API.
+ */
+public class MicrosoftCalendarExporter implements Exporter<TokenAuthData, CalendarContainerResource> {
+    private static final String CALENDARS_URL = "/v1.0/me/calendars";
+    private static final String EVENTS_URL = "/v1.0/me/calendars/%s/events";
+    private static final String ODATA_NEXT = "@odata.nextLink";
+
+    private final String baseUrl;
+
+    private final OkHttpClient client;
+    private final ObjectMapper objectMapper;
+    private final TransformerService transformerService;
+
+    public MicrosoftCalendarExporter(OkHttpClient client, ObjectMapper objectMapper, TransformerService transformerService) {
+        this("https://graph.microsoft.com/", client, objectMapper, transformerService);
+    }
+
+    public MicrosoftCalendarExporter(String baseUrl, OkHttpClient client, ObjectMapper objectMapper, TransformerService transformerService) {
```

Comment:
Is this only for testing? If so, we can add @ VisibleForTesting

Reply:
OK